src/state/selectors/index.js

```diff
@@ -1,10 +1,37 @@
+import { createSelector } from 'reselect';
 import filter from 'lodash/filter';
 import flatten from 'lodash/flatten';
 import { LanguageMap } from 'manifesto.js';
 import Annotation from '../../lib/Annotation';
 import ManifestoCanvas from '../../lib/ManifestoCanvas';
 import CanvasGroupings from '../../lib/CanvasGroupings';
 
+/** */
+const uncurry = curriedFn => (...args) => (
+  args.reduce((left, right) => left(right), curriedFn)
+);
+
+/** */
+const createSelectorN = (...selectors) => {
```

Comment:
Can you give some background for this. I know what currying is and I see what the code does but I cannot reason about why it' needed.

Reply:
You're right, it's unnecessary here (for now). I think it starts to become useful when the selectors start to do more work and/or we end up with selectors with slightly different input (like `getDestructuredMetadata` that acts on either a manifest or canvas), but, yes, let's only add them when the time comes.